camelot/view/controls/delegates/floatdelegate.py

```diff
@@ -65,16 +67,21 @@ def get_standard_item(cls, locale, model_context):
         item.setData(model_context.field_attributes.get('focus_policy'), FocusPolicyRole)
         item.setData(model_context.field_attributes.get('suffix'), SuffixRole)
         item.setData(model_context.field_attributes.get('prefix'), PrefixRole)
-        item.setData(model_context.field_attributes.get('single_step'), SingleStepRole)
+        single_step = model_context.field_attributes.get('single_step')
+        if isinstance(single_step, Decimal):
+            single_step = float(single_step) # FIXME: use bound decimal?
+        item.setData(single_step, SingleStepRole)
         precision = model_context.field_attributes.get('precision', 2)
         item.setData(precision, PrecisionRole)
-        item.setData(minimum, MinimumRole)
-        item.setData(maximum, MaximumRole)
+        item.setData(float(minimum), MinimumRole) # FIXME: use bound decimal?
+        item.setData(float(maximum), MaximumRole) # FIXME: use bound decimal?
         # Set default precision of 2 when precision is undefined, instead of using the default argument of the dictionary's get method,
         # as that only handles the precision key not being present, not it being explicitly set to None.
         if precision is None:
             precision = 2
         if model_context.value is not None:
+            # FIXME: require isinstance(model_context.value, Decimal)?
```

Comment:
ik denk dat deze code beter is, door 'sloppy model programmatie' kunnen hier ook int waardes tussenzitten

worst case zitten er floats tussen ...